akka-cluster-sharding-typed/src/test/scala/akka/cluster/sharding/typed/delivery/ReliableDeliveryShardingSpec.scala

```diff
@@ -407,6 +408,7 @@ class ReliableDeliveryShardingSpec
             // It is possible the ProducerController re-sends msg-3 again before it has processed its acknowledgement.
             // If the ConsumerController restarts between sending the acknowledgement and receiving that re-sent msg-3,
             // it will deliver msg-3 a second time. We then expect msg-4 next:
+            delivery3cor4.confirmTo ! ConsumerController.Confirmed
             val delivery4 = consumerProbes(2).receiveMessage()
             delivery4.message should ===(TestConsumer.Job("msg-4"))
```

Comment:
The test failed here. There are a lot of details in the logs but after analyzing them carefully it's clear that msg-4 isn't delivered because of the missing Confirmed of msg-3.

Logs with some `>>>` comments:
```

2022-10-11T01:18:52.8621569Z 2022-10-11 01:18:44,655 INFO  akka.actor.testkit.typed.scaladsl.LogCapturing  - Logging started for test [akka.cluster.sharding.typed.delivery.ReliableDeliveryShardingSpec: ReliableDelivery with sharding must deliver unconfirmed if ShardingConsumerController is terminated] MDC: {}
2022-10-11T01:18:52.8623286Z 2022-10-11 01:18:44,657 INFO  akka.cluster.sharding.typed.scaladsl.ClusterSharding  - Starting Shard Region [TestConsumer-6]... MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaUid=-7107222847756677322, sourceThread=pool-1-thread-1-ScalaTest-running-ReliableDeliveryShardingSpec, akkaSource=ClusterSharding(akka://ReliableDeliveryShardingSpec), sourceActorSystem=ReliableDeliveryShardingSpec, akkaTimestamp=01:18:44.657UTC}
2022-10-11T01:18:52.8624970Z 2022-10-11 01:18:44,658 INFO  akka.cluster.sharding.ShardRegion  - TestConsumer-6: Automatic entity passivation: idle entities after [2.000 min], checked every [1.000 min] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaUid=-7107222847756677322, sourceThread=ReliableDeliveryShardingSpec-akka.actor.internal-dispatcher-7, akkaSource=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013/system/sharding/TestConsumer-6, sourceActorSystem=ReliableDeliveryShardingSpec, akkaTimestamp=01:18:44.658UTC}
2022-10-11T01:18:52.8626942Z 2022-10-11 01:18:44,660 INFO  akka.cluster.singleton.ClusterSingletonManager akkaClusterSingletonStarted - Singleton manager starting singleton actor [akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6Coordinator/singleton] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaUid=-7107222847756677322, sourceThread=ReliableDeliveryShardingSpec-akka.actor.internal-dispatcher-8, akkaSource=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013/system/sharding/TestConsumer-6Coordinator, sourceActorSystem=ReliableDeliveryShardingSpec, akkaTimestamp=01:18:44.660UTC}
2022-10-11T01:18:52.8628608Z 2022-10-11 01:18:44,660 INFO  akka.cluster.singleton.ClusterSingletonManager  - ClusterSingletonManager state change [Start -> Oldest] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaUid=-7107222847756677322, sourceThread=ReliableDeliveryShardingSpec-akka.actor.internal-dispatcher-8, akkaSource=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013/system/sharding/TestConsumer-6Coordinator, sourceActorSystem=ReliableDeliveryShardingSpec, akkaTimestamp=01:18:44.660UTC}
2022-10-11T01:18:52.8629851Z 2022-10-11 01:18:44,661 DEBUG akka.cluster.sharding.typed.delivery.ShardingProducerController  - Creating ProducerController for entity [entity-1] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
2022-10-11T01:18:52.8631611Z 2022-10-11 01:18:44,661 INFO  akka.cluster.sharding.DDataShardCoordinator  - TestConsumer-6: ShardCoordinator was moved to the active state with [0] shards MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaUid=-7107222847756677322, sourceThread=ReliableDeliveryShardingSpec-akka.actor.default-dispatcher-16, akkaSource=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013/system/sharding/TestConsumer-6Coordinator/singleton/coordinator, sourceActorSystem=ReliableDeliveryShardingSpec, akkaTimestamp=01:18:44.661UTC}
2022-10-11T01:18:52.8632868Z 2022-10-11 01:18:44,661 TRACE akka.cluster.sharding.typed.delivery.ShardingProducerController  - Received RequestNext from [entity-1], confirmed seqNr [0] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}

>>> msg-1
2022-10-11T01:18:52.8634246Z 2022-10-11 01:18:44,661 TRACE akka.cluster.sharding.typed.delivery.ShardingProducerController  - Sending [akka.actor.typed.delivery.TestConsumer$Job] to [p-6-entity-1] with outSeqNr [1]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, producerId=p-6, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8635599Z 2022-10-11 01:18:44,661 TRACE akka.actor.typed.delivery.ProducerController  - Sending [akka.actor.typed.delivery.TestConsumer$Job] with seqNr [1]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8636862Z 2022-10-11 01:18:44,931 DEBUG akka.cluster.sharding.typed.delivery.ShardingConsumerController  - Starting ConsumerController for producerId [p-6-entity-1]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8638062Z 2022-10-11 01:18:44,932 DEBUG akka.actor.typed.delivery.ConsumerController  - Received Start, unstash [0] messages. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
>> delivering msg-1
2022-10-11T01:18:52.8639435Z 2022-10-11 01:18:44,932 TRACE akka.actor.typed.delivery.ConsumerController  - Received first SequencedMessage seqNr [1], delivering to consumer. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8641011Z 2022-10-11 01:18:44,932 DEBUG akka.actor.typed.delivery.ConsumerController  - Associated with new ProducerController [Actor[akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1#491821914]], seqNr [1]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, producerId=p-6-entity-1, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8642394Z 2022-10-11 01:18:44,932 DEBUG akka.actor.typed.delivery.ConsumerController  - Sending Request with requestUpToSeqNr [50] after first SequencedMessage. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, producerId=p-6-entity-1, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8643598Z 2022-10-11 01:18:44,932 DEBUG akka.actor.typed.delivery.ProducerController  - Received Request, confirmed [0], requested [50], current [2] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8644864Z 2022-10-11 01:18:44,932 TRACE akka.cluster.sharding.typed.delivery.ShardingProducerController  - Received RequestNext from [entity-1], confirmed seqNr [0] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
2022-10-11T01:18:52.8646134Z 2022-10-11 01:18:44,933 TRACE akka.actor.typed.delivery.ConsumerController  - Received Confirmed seqNr [1] from consumer, stashed size [0]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8647424Z 2022-10-11 01:18:44,933 DEBUG akka.actor.typed.delivery.ConsumerController  - Sending Request after first with confirmedSeqNr [1], requestUpToSeqNr [50]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}

>>> msg-2
2022-10-11T01:18:52.8648903Z 2022-10-11 01:18:44,933 TRACE akka.cluster.sharding.typed.delivery.ShardingProducerController  - Sending [akka.actor.typed.delivery.TestConsumer$Job] to [p-6-entity-1] with outSeqNr [2]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
2022-10-11T01:18:52.8650093Z 2022-10-11 01:18:44,933 DEBUG akka.actor.typed.delivery.ProducerController  - Received Request, confirmed [1], requested [50], current [2] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8651281Z 2022-10-11 01:18:44,933 TRACE akka.actor.typed.delivery.ProducerController  - Sending confirmation replies from [1] to [1]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, producerId=p-6-entity-1, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8652521Z 2022-10-11 01:18:44,933 TRACE akka.actor.typed.delivery.ProducerController  - Sending [akka.actor.typed.delivery.TestConsumer$Job] with seqNr [2]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8653735Z 2022-10-11 01:18:44,933 TRACE akka.cluster.sharding.typed.delivery.ShardingProducerController  - Received Ack, confirmed [1], current [1]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
2022-10-11T01:18:52.8654986Z 2022-10-11 01:18:44,933 TRACE akka.cluster.sharding.typed.delivery.ShardingProducerController  - Received RequestNext from [entity-1], confirmed seqNr [1] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
>>> delivering msg-2
2022-10-11T01:18:52.8656315Z 2022-10-11 01:18:44,933 TRACE akka.actor.typed.delivery.ConsumerController  - Received SequencedMessage seqNr [2], delivering to consumer. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8657862Z 2022-10-11 01:18:44,933 TRACE akka.actor.typed.delivery.ConsumerController  - Received Confirmed seqNr [2] from consumer, stashed size [0]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8659019Z 2022-10-11 01:18:44,933 TRACE akka.actor.typed.delivery.ConsumerController  - Sending Ack seqNr [2]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8660165Z 2022-10-11 01:18:44,933 TRACE akka.actor.typed.delivery.ProducerController  - Received Ack, confirmed [2], current [3]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8661460Z 2022-10-11 01:18:44,933 TRACE akka.actor.typed.delivery.ProducerController  - Sending confirmation replies from [2] to [2]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, producerId=p-6-entity-1, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}

>>> msg-3
2022-10-11T01:18:52.8662841Z 2022-10-11 01:18:44,933 TRACE akka.cluster.sharding.typed.delivery.ShardingProducerController  - Sending [akka.actor.typed.delivery.TestConsumer$Job] to [p-6-entity-1] with outSeqNr [3]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
2022-10-11T01:18:52.8664057Z 2022-10-11 01:18:44,933 TRACE akka.cluster.sharding.typed.delivery.ShardingProducerController  - Received Ack, confirmed [2], current [1]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
2022-10-11T01:18:52.8665319Z 2022-10-11 01:18:44,933 TRACE akka.actor.typed.delivery.ProducerController  - Sending [akka.actor.typed.delivery.TestConsumer$Job] with seqNr [3]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8666564Z 2022-10-11 01:18:44,933 TRACE akka.cluster.sharding.typed.delivery.ShardingProducerController  - Received RequestNext from [entity-1], confirmed seqNr [2] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
>>> delivering msg-3
2022-10-11T01:18:52.8667911Z 2022-10-11 01:18:44,933 TRACE akka.actor.typed.delivery.ConsumerController  - Received SequencedMessage seqNr [3], delivering to consumer. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}

>>> terminated first time
2022-10-11T01:18:52.8669077Z 2022-10-11 01:18:44,934 DEBUG akka.cluster.sharding.typed.delivery.ShardingConsumerController  - Consumer terminated. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
>>> producer has confirmed 2, consumer has not processed 3
2022-10-11T01:18:52.8670228Z 2022-10-11 01:18:44,934 TRACE akka.actor.typed.delivery.ProducerController  - Received Ack, confirmed [2], current [4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}


>>> msg-4
2022-10-11T01:18:52.8671592Z 2022-10-11 01:18:45,934 TRACE akka.cluster.sharding.typed.delivery.ShardingProducerController  - Sending [akka.actor.typed.delivery.TestConsumer$Job] to [p-6-entity-1] with outSeqNr [4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
2022-10-11T01:18:52.8672847Z 2022-10-11 01:18:45,934 TRACE akka.actor.typed.delivery.ProducerController  - Sending [akka.actor.typed.delivery.TestConsumer$Job] with seqNr [4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8674082Z 2022-10-11 01:18:45,934 TRACE akka.cluster.sharding.typed.delivery.ShardingProducerController  - Received RequestNext from [entity-1], confirmed seqNr [2] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
2022-10-11T01:18:52.8675476Z 2022-10-11 01:18:45,935 DEBUG akka.cluster.sharding.typed.delivery.ShardingConsumerController  - Starting ConsumerController for producerId [p-6-entity-1]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8676670Z 2022-10-11 01:18:45,935 DEBUG akka.actor.typed.delivery.ConsumerController  - Received Start, unstash [0] messages. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8678369Z 2022-10-11 01:18:45,935 DEBUG akka.actor.typed.delivery.ConsumerController  - Received SequencedMessage seqNr [4], from new producer producer [Actor[akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1#491821914]] but it wasn't first. Resending. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8679465Z 2022-10-11 01:18:45,935 DEBUG akka.actor.typed.delivery.ProducerController  - Resending [3 - 4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8681171Z 2022-10-11 01:18:45,936 DEBUG akka.actor.typed.delivery.ConsumerController  - Received SequencedMessage seqNr [3], from new producer producer [Actor[akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1#491821914]] but it wasn't first. Resending. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8682862Z 2022-10-11 01:18:45,936 DEBUG akka.actor.typed.delivery.ConsumerController  - Received SequencedMessage seqNr [4], from new producer producer [Actor[akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1#491821914]] but it wasn't first. Resending. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8683954Z 2022-10-11 01:18:45,936 DEBUG akka.actor.typed.delivery.ProducerController  - Resending [3 - 4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
>>> delivering msg-3 again
2022-10-11T01:18:52.8685309Z 2022-10-11 01:18:45,936 TRACE akka.actor.typed.delivery.ConsumerController  - Received first SequencedMessage seqNr [3], delivering to consumer. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8686961Z 2022-10-11 01:18:45,936 DEBUG akka.actor.typed.delivery.ConsumerController  - Associated with new ProducerController [Actor[akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1#491821914]], seqNr [3]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, producerId=p-6-entity-1, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8688378Z 2022-10-11 01:18:45,936 DEBUG akka.actor.typed.delivery.ConsumerController  - Sending Request with requestUpToSeqNr [52] after first SequencedMessage. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, producerId=p-6-entity-1, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8689827Z 2022-10-11 01:18:45,936 TRACE akka.actor.typed.delivery.ConsumerController  - Received SequencedMessage seqNr [4], stashing while waiting for consumer to confirm [3], stashed size [1]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8691087Z 2022-10-11 01:18:45,936 TRACE akka.actor.typed.delivery.ConsumerController  - Received Confirmed seqNr [3] from consumer, stashed size [1]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8692399Z 2022-10-11 01:18:45,936 DEBUG akka.actor.typed.delivery.ConsumerController  - Sending Request after first with confirmedSeqNr [3], requestUpToSeqNr [52]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
>>> delivering msg-4 from stash
2022-10-11T01:18:52.8693672Z 2022-10-11 01:18:45,936 TRACE akka.actor.typed.delivery.ConsumerController  - Received SequencedMessage seqNr [4], delivering to consumer. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}

>>> terminated second time
2022-10-11T01:18:52.8694821Z 2022-10-11 01:18:45,937 DEBUG akka.cluster.sharding.typed.delivery.ShardingConsumerController  - Consumer terminated. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8695917Z 2022-10-11 01:18:45,939 DEBUG akka.actor.typed.delivery.ProducerController  - Resending [3 - 4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8697200Z 2022-10-11 01:18:45,939 DEBUG akka.actor.typed.delivery.ProducerController  - Received Request, confirmed [0], requested [52], current [5] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8698404Z 2022-10-11 01:18:45,939 DEBUG akka.actor.typed.delivery.ProducerController  - Received Request, confirmed [3], requested [52], current [5] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8699679Z 2022-10-11 01:18:45,939 TRACE akka.actor.typed.delivery.ProducerController  - Sending confirmation replies from [3] to [3]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, producerId=p-6-entity-1, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8700940Z 2022-10-11 01:18:45,939 TRACE akka.cluster.sharding.typed.delivery.ShardingProducerController  - Received Ack, confirmed [3], current [1]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
2022-10-11T01:18:52.8702209Z 2022-10-11 01:18:45,940 DEBUG akka.cluster.sharding.typed.delivery.ShardingConsumerController  - Starting ConsumerController for producerId [p-6-entity-1]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8703412Z 2022-10-11 01:18:45,940 DEBUG akka.actor.typed.delivery.ConsumerController  - Received Start, unstash [0] messages. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
>>> delivering msg-3 again
2022-10-11T01:18:52.8704769Z 2022-10-11 01:18:45,940 TRACE akka.actor.typed.delivery.ConsumerController  - Received first SequencedMessage seqNr [3], delivering to consumer. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}

>>> below is missing "Received Confirmed seqNr [3] from consumer" and therefore the expected msg-4 is not delivered
2022-10-11T01:18:52.8706339Z 2022-10-11 01:18:45,940 DEBUG akka.actor.typed.delivery.ConsumerController  - Associated with new ProducerController [Actor[akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1#491821914]], seqNr [3]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, producerId=p-6-entity-1, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8707710Z 2022-10-11 01:18:45,940 DEBUG akka.actor.typed.delivery.ConsumerController  - Sending Request with requestUpToSeqNr [52] after first SequencedMessage. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, producerId=p-6-entity-1, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec}
2022-10-11T01:18:52.8709150Z 2022-10-11 01:18:45,940 TRACE akka.actor.typed.delivery.ConsumerController  - Received SequencedMessage seqNr [4], stashing while waiting for consumer to confirm [3], stashed size [1]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8710318Z 2022-10-11 01:18:45,940 TRACE akka.actor.typed.delivery.ProducerController  - Received Ack, confirmed [3], current [5]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8711503Z 2022-10-11 01:18:45,940 DEBUG akka.actor.typed.delivery.ProducerController  - Received Request, confirmed [0], requested [52], current [5] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8712919Z 2022-10-11 01:18:47,706 DEBUG akka.cluster.sharding.typed.delivery.ShardingProducerController  - Resend first unconfirmed for [p-6-entity-1], because it was idle for [1766 ms] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
2022-10-11T01:18:52.8714045Z 2022-10-11 01:18:47,710 DEBUG akka.actor.typed.delivery.ProducerController  - Resending first unconfirmed [4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8715338Z 2022-10-11 01:18:47,710 DEBUG akka.actor.typed.delivery.ConsumerController  - Received duplicate SequencedMessage seqNr [4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8716662Z 2022-10-11 01:18:48,467 DEBUG akka.cluster.sharding.typed.delivery.ShardingProducerController  - Resend first unconfirmed for [p-6-entity-1], because it was idle for [2527 ms] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
2022-10-11T01:18:52.8717798Z 2022-10-11 01:18:48,467 DEBUG akka.actor.typed.delivery.ProducerController  - Resending first unconfirmed [4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8719105Z 2022-10-11 01:18:48,467 DEBUG akka.actor.typed.delivery.ConsumerController  - Received duplicate SequencedMessage seqNr [4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8720412Z 2022-10-11 01:18:49,226 DEBUG akka.cluster.sharding.typed.delivery.ShardingProducerController  - Resend first unconfirmed for [p-6-entity-1], because it was idle for [3286 ms] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
2022-10-11T01:18:52.8721564Z 2022-10-11 01:18:49,226 DEBUG akka.actor.typed.delivery.ProducerController  - Resending first unconfirmed [4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8722865Z 2022-10-11 01:18:49,226 DEBUG akka.actor.typed.delivery.ConsumerController  - Received duplicate SequencedMessage seqNr [4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8724171Z 2022-10-11 01:18:49,986 DEBUG akka.cluster.sharding.typed.delivery.ShardingProducerController  - Resend first unconfirmed for [p-6-entity-1], because it was idle for [4047 ms] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
2022-10-11T01:18:52.8725378Z 2022-10-11 01:18:49,987 DEBUG akka.actor.typed.delivery.ProducerController  - Resending first unconfirmed [4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8726729Z 2022-10-11 01:18:49,987 DEBUG akka.actor.typed.delivery.ConsumerController  - Received duplicate SequencedMessage seqNr [4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8728052Z 2022-10-11 01:18:50,746 DEBUG akka.cluster.sharding.typed.delivery.ShardingProducerController  - Resend first unconfirmed for [p-6-entity-1], because it was idle for [4806 ms] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
2022-10-11T01:18:52.8729190Z 2022-10-11 01:18:50,746 DEBUG akka.actor.typed.delivery.ProducerController  - Resending first unconfirmed [4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8730481Z 2022-10-11 01:18:50,746 DEBUG akka.actor.typed.delivery.ConsumerController  - Received duplicate SequencedMessage seqNr [4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8731783Z 2022-10-11 01:18:51,506 DEBUG akka.cluster.sharding.typed.delivery.ShardingProducerController  - Resend first unconfirmed for [p-6-entity-1], because it was idle for [5566 ms] MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6}
2022-10-11T01:18:52.8732933Z 2022-10-11 01:18:51,506 DEBUG akka.actor.typed.delivery.ProducerController  - Resending first unconfirmed [4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/user/shardingController-6/entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8734206Z 2022-10-11 01:18:51,506 DEBUG akka.actor.typed.delivery.ConsumerController  - Received duplicate SequencedMessage seqNr [4]. MDC: {akkaAddress=akka://ReliableDeliveryShardingSpec@10.1.0.81:36013, akkaSource=akka://ReliableDeliveryShardingSpec/system/sharding/TestConsumer-6/793/entity-1/consumerController-p-6-entity-1, sourceActorSystem=ReliableDeliveryShardingSpec, producerId=p-6-entity-1}
2022-10-11T01:18:52.8735516Z 2022-10-11 01:18:51,941 INFO  akka.actor.testkit.typed.scaladsl.LogCapturing  - Logging finished for test [akka.cluster.sharding.typed.delivery.ReliableDeliveryShardingSpec: ReliableDelivery with sharding must deliver unconfirmed if ShardingConsumerController is terminated] that [Failed(java.lang.AssertionError: Timeout (6 seconds) during receiveMessage while waiting for message.)] MDC: {}
2022-10-11T01:18:52.8739790Z <-- [[34makka.cluster.sharding.typed.delivery.ReliableDeliveryShardingSpec: ReliableDelivery with sharding must deliver unconfirmed if ShardingConsumerController is terminated[0m] End of log messages of test that [Failed(java.lang.AssertionError: Timeout (6 seconds) during receiveMessage while waiting for message.)]
2022-10-11T01:18:52.8740721Z [10-11 01:18:52.779] [0m[[0m[0minfo[0m] [0m[0m[31m- must deliver unconfirmed if ShardingConsumerController is terminated *** FAILED *** (8 seconds, 88 milliseconds)[0m[0m
2022-10-11T01:18:52.8741762Z [10-11 01:18:52.779] [0m[[0m[0minfo[0m] [0m[0m[31m  java.lang.AssertionError: Timeout (6 seconds) during receiveMessage while waiting for message.[0m[0m
2022-10-11T01:18:52.8742744Z [10-11 01:18:52.779] [0m[[0m[0minfo[0m] [0m[0m[31m  at akka.actor.testkit.typed.internal.TestProbeImpl.assertFail(TestProbeImpl.scala:399)[0m[0m
2022-10-11T01:18:52.8743693Z [10-11 01:18:52.779] [0m[[0m[0minfo[0m] [0m[0m[31m  at akka.actor.testkit.typed.internal.TestProbeImpl.$anonfun$receiveMessage_internal$1(TestProbeImpl.scala:182)[0m[0m
2022-10-11T01:18:52.8744392Z [10-11 01:18:52.779] [0m[[0m[0minfo[0m] [0m[0m[31m  at scala.Option.getOrElse(Option.scala:201)[0m[0m
2022-10-11T01:18:52.8745315Z [10-11 01:18:52.779] [0m[[0m[0minfo[0m] [0m[0m[31m  at akka.actor.testkit.typed.internal.TestProbeImpl.receiveMessage_internal(TestProbeImpl.scala:182)[0m[0m
2022-10-11T01:18:52.8746217Z [10-11 01:18:52.779] [0m[[0m[0minfo[0m] [0m[0m[31m  at akka.actor.testkit.typed.internal.TestProbeImpl.receiveMessage(TestProbeImpl.scala:175)[0m[0m
2022-10-11T01:18:52.8747266Z [10-11 01:18:52.779] [0m[[0m[0minfo[0m] [0m[0m[31m  at akka.cluster.sharding.typed.delivery.ReliableDeliveryShardingSpec.$anonfun$new$15(ReliableDeliveryShardingSpec.scala:410)[0m[0m
2022-10-11T01:18:52.8748048Z [10-11 01:18:52.780] [0m[[0m[0minfo[0m] [0m[0m[31m  at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)[0m[0m
2022-10-11T01:18:52.8748803Z [10-11 01:18:52.780] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)[0m[0m
2022-10-11T01:18:52.8749551Z [10-11 01:18:52.780] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)[0m[0m
2022-10-11T01:18:52.8750280Z [10-11 01:18:52.780] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)[0m[0m
2022-10-11T01:18:52.8751019Z [10-11 01:18:52.780] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.Transformer.apply(Transformer.scala:22)[0m[0m
2022-10-11T01:18:52.8751756Z [10-11 01:18:52.780] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.Transformer.apply(Transformer.scala:20)[0m[0m
2022-10-11T01:18:52.8752602Z [10-11 01:18:52.780] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.wordspec.AnyWordSpecLike$$anon$3.apply(AnyWordSpecLike.scala:1076)[0m[0m
2022-10-11T01:18:52.8753467Z [10-11 01:18:52.780] [0m[[0m[0minfo[0m] [0m[0m[31m  at akka.actor.testkit.typed.scaladsl.LogCapturing.withFixture(LogCapturing.scala:70)[0m[0m
2022-10-11T01:18:52.8754339Z [10-11 01:18:52.780] [0m[[0m[0minfo[0m] [0m[0m[31m  at akka.actor.testkit.typed.scaladsl.LogCapturing.withFixture$(LogCapturing.scala:68)[0m[0m
2022-10-11T01:18:52.8755394Z [10-11 01:18:52.780] [0m[[0m[0minfo[0m] [0m[0m[31m  at akka.cluster.sharding.typed.delivery.ReliableDeliveryShardingSpec.withFixture(ReliableDeliveryShardingSpec.scala:91)[0m[0m
2022-10-11T01:18:52.8756286Z [10-11 01:18:52.780] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.wordspec.AnyWordSpecLike.invokeWithFixture$1(AnyWordSpecLike.scala:1074)[0m[0m
2022-10-11T01:18:52.8757205Z [10-11 01:18:52.781] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$runTest$1(AnyWordSpecLike.scala:1086)[0m[0m
2022-10-11T01:18:52.8757975Z [10-11 01:18:52.781] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)[0m[0m
2022-10-11T01:18:52.8758809Z [10-11 01:18:52.781] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.wordspec.AnyWordSpecLike.runTest(AnyWordSpecLike.scala:1086)[0m[0m
2022-10-11T01:18:52.8759642Z [10-11 01:18:52.781] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.wordspec.AnyWordSpecLike.runTest$(AnyWordSpecLike.scala:1068)[0m[0m
2022-10-11T01:18:52.8760667Z [10-11 01:18:52.781] [0m[[0m[0minfo[0m] [0m[0m[31m  at akka.cluster.sharding.typed.delivery.ReliableDeliveryShardingSpec.runTest(ReliableDeliveryShardingSpec.scala:91)[0m[0m
2022-10-11T01:18:52.8761613Z [10-11 01:18:52.781] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$runTests$1(AnyWordSpecLike.scala:1145)[0m[0m
2022-10-11T01:18:52.8762491Z [10-11 01:18:52.781] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)[0m[0m
2022-10-11T01:18:52.8763177Z [10-11 01:18:52.781] [0m[[0m[0minfo[0m] [0m[0m[31m  at scala.collection.immutable.List.foreach(List.scala:333)[0m[0m
2022-10-11T01:18:52.8765164Z [10-11 01:18:52.781] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)[0m[0m
2022-10-11T01:18:52.8766243Z [10-11 01:18:52.781] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:390)[0m[0m
2022-10-11T01:18:52.8767019Z [10-11 01:18:52.781] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:427)[0m[0m
2022-10-11T01:18:52.8767700Z [10-11 01:18:52.781] [0m[[0m[0minfo[0m] [0m[0m[31m  at scala.collection.immutable.List.foreach(List.scala:333)[0m[0m
2022-10-11T01:18:52.8768385Z [10-11 01:18:52.782] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)[0m[0m
2022-10-11T01:18:52.8769051Z [10-11 01:18:52.782] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)[0m[0m
2022-10-11T01:18:52.8769707Z [10-11 01:18:52.782] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)[0m[0m
2022-10-11T01:18:52.8770462Z [10-11 01:18:52.782] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.wordspec.AnyWordSpecLike.runTests(AnyWordSpecLike.scala:1145)[0m[0m
2022-10-11T01:18:52.8771190Z [10-11 01:18:52.782] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.wordspec.AnyWordSpecLike.runTests$(AnyWordSpecLike.scala:1144)[0m[0m
2022-10-11T01:18:52.8772179Z [10-11 01:18:52.782] [0m[[0m[0minfo[0m] [0m[0m[31m  at akka.cluster.sharding.typed.delivery.ReliableDeliveryShardingSpec.runTests(ReliableDeliveryShardingSpec.scala:91)[0m[0m
2022-10-11T01:18:52.8773043Z [10-11 01:18:52.782] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.Suite.run(Suite.scala:1112)[0m[0m
2022-10-11T01:18:52.8773678Z [10-11 01:18:52.782] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.Suite.run$(Suite.scala:1094)[0m[0m
2022-10-11T01:18:52.8774687Z [10-11 01:18:52.782] [0m[[0m[0minfo[0m] [0m[0m[31m  at akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit.org$scalatest$BeforeAndAfterAll$$super$run(ScalaTestWithActorTestKit.scala:31)[0m[0m
2022-10-11T01:18:52.8775439Z [10-11 01:18:52.782] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)[0m[0m
2022-10-11T01:18:52.8776122Z [10-11 01:18:52.782] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)[0m[0m
2022-10-11T01:18:52.8776940Z [10-11 01:18:52.782] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)[0m[0m
2022-10-11T01:18:52.8778054Z [10-11 01:18:52.783] [0m[[0m[0minfo[0m] [0m[0m[31m  at akka.cluster.sharding.typed.delivery.ReliableDeliveryShardingSpec.org$scalatest$wordspec$AnyWordSpecLike$$super$run(ReliableDeliveryShardingSpec.scala:91)[0m[0m
2022-10-11T01:18:52.8778785Z [10-11 01:18:52.783] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$run$1(AnyWordSpecLike.scala:1190)[0m[0m
2022-10-11T01:18:52.8779405Z [10-11 01:18:52.783] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.SuperEngine.runImpl(Engine.scala:535)[0m[0m
2022-10-11T01:18:52.8780108Z [10-11 01:18:52.783] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.wordspec.AnyWordSpecLike.run(AnyWordSpecLike.scala:1190)[0m[0m
2022-10-11T01:18:52.8781211Z [10-11 01:18:52.783] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.wordspec.AnyWordSpecLike.run$(AnyWordSpecLike.scala:1188)[0m[0m
2022-10-11T01:18:52.8782152Z [10-11 01:18:52.784] [0m[[0m[0minfo[0m] [0m[0m[31m  at akka.cluster.sharding.typed.delivery.ReliableDeliveryShardingSpec.run(ReliableDeliveryShardingSpec.scala:91)[0m[0m
2022-10-11T01:18:52.8783019Z [10-11 01:18:52.784] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:318)[0m[0m
2022-10-11T01:18:52.8783734Z [10-11 01:18:52.784] [0m[[0m[0minfo[0m] [0m[0m[31m  at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:513)[0m[0m
2022-10-11T01:18:52.8784320Z [10-11 01:18:52.784] [0m[[0m[0minfo[0m] [0m[0m[31m  at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:413)[0m[0m
2022-10-11T01:18:52.8784987Z [10-11 01:18:52.784] [0m[[0m[0minfo[0m] [0m[0m[31m  at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)[0m[0m
2022-10-11T01:18:52.8785806Z [10-11 01:18:52.784] [0m[[0m[0minfo[0m] [0m[0m[31m  at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)[0m[0m
2022-10-11T01:18:52.8786559Z [10-11 01:18:52.784] [0m[[0m[0minfo[0m] [0m[0m[31m  at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)[0m[0m
2022-10-11T01:18:52.8787140Z [10-11 01:18:52.784] [0m[[0m[0minfo[0m] [0m[0m[31m  at java.base/java.lang.Thread.run(Thread.java:833)[0m[0m
2022-10-11T01:18:55.0689948Z [10-11 01:18:55.068] [0m[[0m[0minfo[0m] [0m[0m[32m- must cleanup unused ProducerController (2 seconds, 281 milliseconds)[0m[0m
2022-10-11T01:18:55.4606303Z [10-11 01:18:55.460] [0m[[0m[0minfo[0m] [0m[0m[32m- must cleanup ConsumerController when ProducerController is terminated (390 milliseconds)[0m[0m
```